lib/axon.ex

```diff
@@ -2387,6 +2390,130 @@ defmodule Axon do
     end
   end
 
+  ## Serialization
+
+  @doc """
+  Serializes a model and it's parameters for persisting
+  models to disk or elsewhere.
+
+  Model and parameters are serialized as a tuple, where the
+  model is converted to a recursive map to ensure compatibility
+  with future Axon versions and the parameters are serialized
+  using `Nx.serialize`. There is some additional metadata included
+  such as current serialization version for compatibility.
+
+  Serialization `opts` are forwarded to Nx.serialize and
+  `:erlang.term_to_binary` for controlling compression options.
+
+  ## Examples
+
+      iex> model = Axon.input({nil, 2}) |> Axon.dense(1, kernel_initializer: :zeros, activation: :relu)
+      iex> params = Axon.init(model)
+      iex> serialized = Axon.serialize(model, params)
+      iex> {saved_model, saved_params} = Axon.deserialize(serialized)
+      iex> saved_model
```

Comment:
Perhaps there could be a test in which you `assert_all_close` on each layer, similarly to the `saved_params` assertions below, having used an orthogonal layer initialization or any other non-constant tensor value.
This would not be a doctest, though.